xla/backends/profiler/gpu/cupti_tracer.h

```diff
@@ -31,6 +32,29 @@ limitations under the License.
 namespace xla {
 namespace profiler {
 
+// Provide safe types if CUPTI_PM_SAMPLING is not defined
+// (And therefor CUPTI PM sampling headers are not included)
+#if CUPTI_PM_SAMPLING == 0
+typedef int CUpti_PmSampling_DecodeStopReason;
+constexpr CUpti_PmSampling_DecodeStopReason
+    CUPTI_PM_SAMPLING_DECODE_STOP_REASON_COUNT = 0;
+#endif
+
+// Information related to a decode counters pass over a single device
+struct PmSamplingDecodeInfo {
+  CUpti_PmSampling_DecodeStopReason decode_stop_reason =
+      CUPTI_PM_SAMPLING_DECODE_STOP_REASON_COUNT;
+  uint8_t overflow_ = 0;
+  size_t num_samples = 0;
+  size_t num_completed = 0;
+  size_t num_populated = 0;
+  int device_id;
+  std::vector<SamplerRange> sampler_ranges;
+  std::vector<char const*> metrics;
```

Comment:
```suggestion
  std::vector<std::string> metrics;
```
Let's use std::string here it makes it easier to reason about ownership.